sentry-apache-http-client-5/src/main/java/io/sentry/transport/apache/ApacheHttpClientTransport.java

```diff
@@ -37,28 +40,39 @@
   private final @NotNull RequestDetails requestDetails;
   private final @NotNull CloseableHttpAsyncClient httpclient;
   private final @NotNull RateLimiter rateLimiter;
+  private final @NotNull ClientReportRecorder clientReportRecorder;
   private final @NotNull ReusableCountLatch currentlyRunning;
 
   public ApacheHttpClientTransport(
       final @NotNull SentryOptions options,
       final @NotNull RequestDetails requestDetails,
       final @NotNull CloseableHttpAsyncClient httpclient,
-      final @NotNull RateLimiter rateLimiter) {
-    this(options, requestDetails, httpclient, rateLimiter, new ReusableCountLatch());
+      final @NotNull RateLimiter rateLimiter,
+      final @NotNull ClientReportRecorder clientReportRecorder) {
+    this(
+        options,
+        requestDetails,
+        httpclient,
+        rateLimiter,
+        clientReportRecorder,
+        new ReusableCountLatch());
   }
 
   ApacheHttpClientTransport(
       final @NotNull SentryOptions options,
       final @NotNull RequestDetails requestDetails,
       final @NotNull CloseableHttpAsyncClient httpclient,
       final @NotNull RateLimiter rateLimiter,
+      final @NotNull ClientReportRecorder clientReportRecorder,
```

Comment:
Could also use the `ClientReportRecorder` via options but by passing it directly to the constructor it makes testing a bit easier.

Reply:
The downside is that it breaks the Public API, let's discuss if we should take it via ctor or not.

Reply:
It already exists under `options.getClientReportRecorder()` btw.

Reply:
I can also wrap `SentryOptions` to use mocks for testing.

Reply:
[ ] Use ClientReportRecorder via options and adapt tests